ExampleMod/Projectiles/Minions/ExampleSentry.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Microsoft.Xna.Framework;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Projectiles.Minions
+{
+	internal class ExampleSentry : ModProjectile
+	{
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Example Sentry"); //The english name of the projectile
+			Main.projFrames[projectile.type] = 4;
+			ProjectileID.Sets.MinionTargettingFeature[projectile.type] = true;
+		}
+		public override void SetDefaults()
+		{
+			projectile.width = 54;
+			projectile.height = 30;
+			projectile.friendly = true;
+			projectile.magic = true;
+			projectile.sentry = true; //Sets the weapon as a sentry for sentry accessories to properly work.
+			projectile.timeLeft = Projectile.SentryLifeTime;
+			projectile.ignoreWater = true; //If this is set to false, the projectile will be slowed in water.
+			projectile.tileCollide = false;
+			projectile.penetrate = -1;
+		}
+
+		public override Color? GetAlpha(Color lightColor)
+		{
+			//return Color.White;
+			return new Color(255, 255, 255, 0) * (1f - projectile.alpha / 255f);
+		}
+
+		public override void AI()
+		{
+			//This AI will function as a static sentry, and will not move. If you would like to know how to do more advanced minion AI, check out PurityWisp.cs.
+
+			for (int k = 0; k < 1; k++) {
+				int dust = Dust.NewDust(projectile.position + projectile.velocity, projectile.width, projectile.height, 132, projectile.oldVelocity.X * 0.5f, projectile.oldVelocity.Y * 0.5f);
```

Comment:
Turn this into `NewDustPerfect`, and instantiate a `Dust` variable. Then use the dust variable and change it. Makes the code look a lot cleaner.